cmd/checkmetrics/csv.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"math"
+	"os"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/montanaflynn/stats"
+)
+
+// csvRecord holds the data for a single 'test'. Some of the data is imported
+// from the CSV file, and other fields are then calculated from that data
+type csvRecord struct {
+	Records [][]string // The record slices imported from the file
+
+	// All items below here are generated or calculated from the Records
+	ResultStrings []string  // Hold the array of all the Results as strings
+	Results       []float64 // Result array converted to floats
+	Iterations    int       // How many results did we gather
+	Mean          float64   // The 'average'
+	MinVal        float64   // Smallest value we saw
+	MaxVal        float64   // Largest value we saw
+	SD            float64   // Standard Deviation
+	CoV           float64   // Co-efficient of Variation
+}
+
+// load reads in a CSV 'Metrics' results file from the file path given
+func (c *csvRecord) load(filepath string) error {
+	var err error
+	var f *os.File
+	var r *csv.Reader
+
+	log.Debugf("in csv load of [%s]", filepath)
+
+	f, err = os.Open(filepath)
+	if err != nil {
+		log.Warnf("Failed to open csv file [%s]", filepath)
```

Comment:
The `err` (at least sometimes) has less info - such as in this case it says `no such file or directory`, whereas the `Warnf` adds in the name of the actual file. So I think the `Warnf`'s can/do add more context information.
Yes, the log will go to stdout if you don't set the `--log` setting, but the `err` returns don't come out anywhere unless we do it by hand (apart from maybe the one returned out of 'main' I think) - that is, I don't think we have a double-printing issue.

I can see maybe something like `log.Warnf([%s][%s]", filepath, err)` could be more useful actually - I'll go look at the few Warn's there are and contemplate.